vncdotool/rfb.py

```diff
@@ -189,14 +218,28 @@ def _handleVNCAuthResult(self, block):
             self._doClientInitialization()
             return
         elif result == 1:   #failed
-            self.vncAuthFailed("authentication failed")
-            self.transport.loseConnection()
+            if self._version < 3.8:
+                self.vncAuthFailed("authentication failed")
+                self.transport.loseConnection()
+            else:
+                self.expect(self._handleAuthFailed, 4)
         elif result == 2:   #too many
-            self.vncAuthFailed("too many tries to log in")
-            self.transport.loseConnection()
+            if self._version < 3.8:
+                self.vncAuthFailed("too many tries to log in")
+                self.transport.loseConnection()
+            else:
+                self.expect(self._handleAuthFailed, 4)
         else:
             log.msg("unknown auth response (%d)" % result)
 
+    def _handleAuthFailed(self, block):
+        (waitfor,) = unpack("!I", block)
+        self.expect(self._handleAuthMessage, waitfor)
+
+    def _handleAuthMessage(self, block):
```

Comment:
On the surface I find this name confusing, as it only seems to run in the failure case.  Do you have a link to the protocol spec that outlines this flow, as that context might help my understanding.  Otherwise, this PR seems good to me.

Reply:
Yeah, I agree. I was trying to stay consistent with the existing code as much as possible here, specifically `_handleConnMessage` in this case. I think a lot of the methods should be renamed to match the message types defined in the spec, and there are a lot of other small cleanups possible, but I didn't want to mix cosmetic and functional changes.

I'll rename this to `_handleAuthFailedMessage` which is both pretty consistent with the other code and still indicates it's only called for failure.